tests/pkg/ops/tikv.go

```diff
@@ -53,41 +57,54 @@ func (ops *TiKVOps) TruncateSSTFile(opts TruncateOptions) error {
 		})
 	}
 
-	stdout, stderr, err := exec("find", "/var/lib/tikv/db", "-name", "*.sst", "-o", "-name", "*.save")
-	if err != nil {
-		glog.Errorf("list sst files: stderr=%s err=%s", stderr, err.Error())
-		return errors.Annotate(err, "list sst files")
-	}
+	retryCount := 0
+	for ; retryCount < retryLimit; retryCount++ {
+		if retryCount > 0 {
+			time.Sleep(10 * time.Second)
+		}
+		stdout, stderr, err := exec("find", "/var/lib/tikv/db", "-name", "*.sst", "-o", "-name", "*.save")
+		if err != nil {
+			glog.Warningf("list sst files: stderr=%s err=%s", stderr, err.Error())
+			continue
+		}
 
-	sstCandidates := make(map[string]bool)
+		sstCandidates := make(map[string]bool)
 
-	for _, f := range strings.Split(stdout, "\n") {
-		f = strings.TrimSpace(f)
-		if len(f) > 0 {
-			sstCandidates[f] = true
+		for _, f := range strings.Split(stdout, "\n") {
+			f = strings.TrimSpace(f)
+			if len(f) > 0 {
+				sstCandidates[f] = true
+			}
 		}
-	}
 
-	sst := ""
-	for k := range sstCandidates {
-		if strings.HasSuffix(k, ".sst") && !sstCandidates[k+".save"] {
-			sst = k
+		sst := ""
+		for k := range sstCandidates {
+			if strings.HasSuffix(k, ".sst") && !sstCandidates[k+".save"] {
+				sst = k
+			}
+		}
+		if len(sst) == 0 {
+			glog.Warning("cannot find a sst file")
+			continue
 		}
-	}
-	if len(sst) == 0 {
-		return errors.New("cannot find a sst file")
-	}
 
-	_, stderr, err = exec("cp", sst, sst+".save")
-	if err != nil {
-		glog.Errorf("backup sst file: stderr=%s err=%s", stderr, err.Error())
-		return errors.Annotate(err, "backup sst file")
+		_, stderr, err = exec("cp", sst, sst+".save")
+		if err != nil {
+			glog.Warningf("backup sst file: stderr=%s err=%s", stderr, err.Error())
+			continue
+		}
+
+		_, stderr, err = exec("truncate", "-s", "0", sst)
+		if err != nil {
+			glog.Warningf("truncate sst file: stderr=%s err=%s", stderr, err.Error())
+			continue
+		}
+
+		break
 	}
 
-	_, stderr, err = exec("truncate", "-s", "0", sst)
-	if err != nil {
-		glog.Errorf("truncate sst file: stderr=%s err=%s", stderr, err.Error())
-		return errors.Annotate(err, "truncate sst file")
+	if retryCount == retryLimit {
+		return errors.New("failed to truncate sst file after " + strconv.Itoa(retryLimit) + " trials")
```

Comment:
the error log needs  `opts.Namespace` and `opts.Cluster` field

Reply:
I added annotations to log methods. however, it's caller who pass the `opts` arg, that is, the caller must know these info, so I don't think there is a need for adding these fields to returned error.

Reply:
but the caller has not added these fields to error log too
ref: https://github.com/pingcap/tidb-operator/blob/master/tests/failover.go#L74-L81

Reply:
I added some error logs to the case func. If you think every error should be annotated with additional fields, it would be better to create an another PR to do it, because most of methods of `operatorActions` failed to do so.